### PR TITLE
Clarify network-related flags in the master

### PR DESCRIPTION
**Pull request by thockin**
Rename and rejigger flags to make it more obvious what is happening.  Change
the default listen from ChooseHostInterface() to 0.0.0.0.

### CODE REVIEW

**Comment by thockin** [comment]
sketch: e2e is running

@kazegusuri
@roberthbailey 

regarding #6445

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -115,24 +115,28 @@ func NewAPIServer() *APIServer {
...
+		"address must be reachable by the rest of the cluster. If blank, all interfaces will be used.")
+	fs.Var(&s.BindAddress, "public_address_override", "DEPRECATED: see --bind-address instead")
 	fs.IntVar(&s.ReadOnlyPort, "read_only_port", s.ReadOnlyPort, ""+
-		"The port from which to serve read-only resources. If 0, don't serve on a "+
-		"read-only address. It is assumed that firewall rules are set up such that "+
-		"this port is not reachable from outside of the cluster.")
+		"The port on which to serve read-only resources. If 0, don't serve read-only "+
+		"at all. It is assumed that firewall rules are set up such that this port is "+
+		"not reachable from outside of the cluster.")
+	fs.IntVar(&s.SecurePort, "secure_port", s.SecurePort, ""+
```
Comment (vishh) [inline cmd/kube-apiserver/app/server.go]: @thockin: Is there a rationale behind the usage of '-' vs '_' as word separators?

Comment (thockin) [inline cmd/kube-apiserver/app/server.go]: sorry, trnding towards - as the standard.  I can make this consistent for this PR if we agree on content

Comment (vishh) [inline cmd/kube-apiserver/app/server.go]: Should we mark the flags that use '_' as deprecated and create a duplicate flag that uses '-' instead? 
Also that change should happen in a separate PR.

File: cmd/kube-apiserver/app/server.go [hunk 2]
```diff
@@ -115,24 +115,28 @@ func NewAPIServer() *APIServer {
...
+		"the cluster and that port 443 on the cluster's public address is proxied to this "+
 		"port. This is performed by nginx in the default setup.")
-	fs.Var(&s.Address, "address", "The IP address on to serve on (set to 0.0.0.0 for all interfaces)")
-	fs.Var(&s.PublicAddressOverride, "public_address_override", "Public serving address."+
-		"Read only port will be opened on this address, and it is assumed that port "+
-		"443 at this address will be proxied/redirected to '-address':'-port'. If "+
-		"blank, the address in the first listed interface will be used.")
+	fs.IntVar(&s.InsecurePort, "port", s.InsecurePort, "DEPRECATED: see --insecure-port instead")
+	fs.Var(&s.InsecureBindAddress, "insecure-bind-address", ""+
+		"The IP address on which to serve the --insecure-port (set to 0.0.0.0 for all interfaces)")
```
Comment (roberthbailey) [inline cmd/kube-apiserver/app/server.go]: Mention that it defaults to 127.0.0.1

**Comment by roberthbailey** [comment]
Generally LGTM. It's going to suck to have duplicate flag names for a few releases to move from _ to - but I supposed if we want the flags to be standardized we have to endure the pain at some point -- might as well do it now.

**Comment by thockin** [comment]
I am very confused with the test failures - running integration locally works and runs to completion, but running it as part of a build/release.sh does not...

**Comment by roberthbailey** [comment]
I cloned your change locally and 'make release' ran successfully for me.

File: cmd/kube-apiserver/app/server.go (hunk 1 above)
Comment (thockin) [inline cmd/kube-apiserver/app/server.go]: That should be a whole different set of PRs

File: cmd/kube-apiserver/app/server.go (hunk 2 above)
Comment (thockin) [inline cmd/kube-apiserver/app/server.go]: I'm torn - the default gets listed when you run -? but not in docs.  Updated.

**Comment by vishh** [comment]
LGTM. WIll merge once the test pass

**Comment by thockin** [comment]
I can run integration manually and it passes, but building a release fails.  I hear that this may be a larger problem that others are having.

e2e ran and failed on logging errors.

**Comment by thockin** [comment]
e2e passed.  manual integration passes.  Travis and Shippable both fail, but for different reasons

**Comment by vishh** [comment]
Travis run was successful. Merging.

**Comment by kazegusuri** [comment]
Thanks! I confirmed it worked fine.

**Pull request merged**